src/app/patient-dashboard/patient-info/edit-contacts.component.ts

```diff
@@ -0,0 +1,131 @@
+import { Component, OnInit, EventEmitter, Output } from '@angular/core';
+
+import { PatientService } from '../patient.service';
+import { Patient } from '../../models/patient.model';
+import * as _ from 'lodash';
+import { PersonResourceService } from '../../openmrs-api/person-resource.service';
+import { PatientInfoService } from './patient-info.service';
+
+
+@Component({
+  selector: 'edit-contacts-info',
+  templateUrl: 'edit-contacts.component.html',
+  styleUrls: []
+})
+export class EditContactsComponent implements OnInit {
+  public patient: Patient = new Patient({});
+  public display: boolean = false;
+  public patientPhoneNumber: number;
+  public alternativePhoneNumber: number;
+  public patnerPhoneNumber: number;
+  public nextofkinPhoneNumber: number;
+  constructor(private patientService: PatientService,
+              private personResourceService: PersonResourceService,
+              private patientInfoService: PatientInfoService) {
+  }
+  ngOnInit() {
+    this.getPatient();
+  }
+  public showDialog() {
+    this.display = true;
+  }
+  public getPatient() {
+    this.patientService.currentlyLoadedPatient.subscribe(
+      (patient) => {
+        if (patient) {
+          this.patient = patient;
+          this.patnerPhoneNumber = this.patient.person.patnerPhoneNumber;
+          this.patientPhoneNumber = this.patient.person.patientPhoneNumber;
+          this.alternativePhoneNumber =
+            this.patient.person.alternativePhoneNumber;
+          this.nextofkinPhoneNumber = this.patient.person.nextofkinPhoneNumber;
+        }
+      }
+    );
+  }
+  public saveAttribute() {
+    let person = {
+      uuid: this.patient.person.uuid
+      };
+    let personAttributePayload = {
+      attributes: [{
+        value:  this.patientPhoneNumber,
+        attributeType: '72a759a8-1359-11df-a1f1-0026b9348838'
+      }, {
+        value:  this.alternativePhoneNumber,
+        attributeType: 'c725f524-c14a-4468-ac19-4a0e6661c930'
+      }, {
+        value:  this.nextofkinPhoneNumber,
+        attributeType: 'a657a4f1-9c0f-444b-a1fd-445bb91dd12d'
+      }, {
+        value:  this.patnerPhoneNumber,
+        attributeType: 'b0a08406-09c0-4f8b-8cb5-b22b6d4a8e46'
+      }]
+    };
+    let payLoad = [];
+    for (let a of personAttributePayload.attributes) {
```

Comment:
Separate this logic and put it in it's own function. Also this logic requires unit testing.